### DISCUSSION TITLE
CURLOPT_PROXY option in the extension

### DISCUSSION THREAD

**Discussion by GitHub-Ahai** [tectalichq/public-openai-client-php#16] [Q&A] [answered]
I hope that the CURLOPT_PROXY option in the extension package can be freely configured, as some regions require IP proxies to access.

**Comment by thejamescollins** [comment, marked as answer]
Hi there @GitHub-Ahai,

The package allows you to use many different API clients, you can install any PSR-18 compatible HTTP client including Guzzle or the Symfony HTTP Client: https://tectalic.com/apis/openai/docs/php#system-requirements

And Guzzle supports [configuring a proxy for all requests](https://docs.guzzlephp.org/en/stable/request-options.html#proxy) using the `proxy` option.

When you instantiate your Guzzle client, you can set the `proxy` configuration option:

eg:
```php
  $httpClient = new \GuzzleHttp\Client([
        'proxy' => 'tcp://12.34.56.78:3128',
    ]);
```

And then instantiate the OpenAI client:

```php
$auth = new \Tectalic\OpenAi\Authentication(getenv('OPENAI_API_KEY'));
$openaiClient = \Tectalic\OpenAi\Manager::build($httpClient, $auth);
```

And then any requests that you perform using `$openaiClient` should use the specified proxy.

I hope that helps,

James

**Reply by GitHub-Ahai** [reply]
Thank you for your reply
